packages/main/src/plugin/index.ts

```diff
@@ -1101,9 +1101,15 @@ export class PluginSystem {
     this.ipcHandle(
       'container-provider-registry:pushImage',
       async (_listener, engine: string, imageId: string, callbackId: number): Promise<void> => {
-        return containerProviderRegistry.pushImage(engine, imageId, (name: string, data: string) => {
-          this.getWebContentsSender().send('container-provider-registry:pushImage-onData', callbackId, name, data);
-        });
+        const msgName = 'container-provider-registry:pushImage-onData';
+        return containerProviderRegistry
+          .pushImage(engine, imageId, (name: string, data: string) => {
+            this.getWebContentsSender().send(msgName, callbackId, name, data);
+          })
+          .catch((error: unknown) => {
```

Comment:
I don't get why the error is not transfered through the promise and needs this special processing

Reply:
I was not going to improve implementation, I just wanted to make current implementation to work.
The renderer part does not not need full error object, just a text of the error. It is much simpler to send it in form of string 'error' message.

Reply:
Yes but why don't throwing this string instead of implementing a complex exchange mechanism ?

Reply:
Opened #10572 for simpler error processing here.